lib/saml/user_attributes/ssoe.rb

```diff
@@ -50,6 +56,27 @@ def email
       end
 
       ### Identifiers
+      def uuid
+        raise Common::Exceptions::InvalidResource, @attributes unless idme_uuid || sec_id
+
+        return idme_uuid if idme_uuid
+
+        Digest::UUID.uuid_v3('sec-id', sec_id).tr('-', '')
```

Comment:
I got a bit tripped up reading this - is `Digest::UUID.uuid_v3` some kind of convenience way of generating a "uuid-like" hash? And then we're removing the dashes?

Maybe a comment would help. Or maybe wrapping this behavior in a helper method named in a way for added readability?

Reply:
The `sec_id` is not a UUID, and while unique this has a potential to cause issues in downstream processes that are expecting a user UUID to be 32 bytes.  For example, if there is a log filtering process that was striping out any 32 byte id, an 10 byte sec id would be missed.  Using a one way UUID hash, will convert the sec id to a 32 byte unique identifier so that any downstream processes will treat it exactly the same as a typical 32 byte ID.me identifier.

I'm relatively new to the team, so maybe this isn't needed, but I thought this would be the safest route.

@omgitsbillryan if that logic is sound, I will add the above comment or something similar to the code.

NOTE: there is the possibility of a collision between an ID.me UUID and a hashed sec id, however that probability should be so small that I believe it can be ignored

Reply:
Correct, for example the InProgressForm model has some validation around the uuid being a 32 byte uuid. 

I think the UUIDv3 vs. the UUIDv4 (ID.me) algorithms are designed to minimize the possibility of collisions to astronomically low numbers.